ext/tidy/tidy.c

```diff
@@ -809,9 +807,15 @@ static zend_result php_tidy_parse_string(PHPTidyObj *obj, const char *string, ui
 
 	obj->ptdoc->initialized = true;
 
+#ifdef HAVE_TIDYPARSESTRING
+	if (tidyParseString(obj->ptdoc->doc, string) < 0) {
```

Comment:
https://3v4l.org/KMN1s (edit the repro slightly, as tidy ext is available for live/wasm preview only)

I guess this will stop parsing on the first null byte. Currently explicit length is honored.